compiler/deSugar/DsExpr.hs

```diff
@@ -305,6 +305,13 @@ ds_expr _ (HsLamCase _ matches)
   = do { ([discrim_var], matching_code) <- matchWrapper CaseAlt Nothing matches
        ; return $ Lam discrim_var matching_code }
 
+-- Inline bindings extension
+--
+-- Just desugars the inner expression; the actual extension's desugaring is
+-- handled as part of do-notation desugaring (dsDo).
+ds_expr _ (HsInlineBind _ e)
```

Comment:
Either `dsExpr` needs to be extended with a parameter for the current `do` statement context in which we’re accumulating extracted inline bindings, or the `DsM` local environment needs to be extended with this information (something like `IORef [LHsExpr GhcTc]`).